File: CodeForces/101-200/144/A/144A.py
# Problem 144A: Arrival of the General
# https://codeforces.com/contest/144/problem/A
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: September 27th, 2024
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

sol = int(input())
heights = list(map(int, input().split()))
max_index = 0
min_index = 0
for index, height in enumerate(heights):
    if height > heights[max_index]:
        max_index = index
    elif height < heights[min_index]:
        min_index = index
    elif height == heights[min_index]:
        min_index = index

ans = max_index
ans += len(heights)-1-min_index

if max_index > min_index:
    ans -= 1
print(ans)

File: CodeForces/101-200/144/A/144A.rb
# Problem 144A: Arrival of the General
# https://codeforces.com/contest/144/problem/A
# Executed with: ruby 3.3.8 (2025-04-09 revision b200bad6cd) [aarch64-linux-gnu]
# Linux version: Linux kernel 6.17.0-5-generic
# Submitted on: October 26th, 2025
# Codeforces language used: Ruby 3.2.2

_ = gets.chomp
all_positions = gets.split.map(&:to_i)

# Largest to the left
# Smallest to the right

largest_index = -1
largest_value = -1

smallest_index = 9999999
smallest_value = 9999999

for index in 0..all_positions.length-1
  if all_positions[index] > largest_value
    largest_index = index
    largest_value = all_positions[index]
  end

  if all_positions[index] <= smallest_value
    smallest_index = index
    smallest_value = all_positions[index]
  end

end

# If the largest value and smallest value need to pass each other, you move both with one move.
if largest_index > smallest_index
  largest_index -= 1
end

puts largest_index + all_positions.length-smallest_index-1
